src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            try {
+                Long id = (Long) session.save(actor);
+                actor.setId(id);
+                transaction.commit();
+                return actor;
+            } catch (Exception e) {
+                transaction.rollback();
+                throw new DataProcessingException("Error while adding an actor", e);
+            }
+        } catch (Exception e) {
+            throw new DataProcessingException("Error while opening a session", e);
+        }
```

Comment:
using an inner try-catch statement extremely reduces the readability of your code. 

Can be refactored with only one try-catch.